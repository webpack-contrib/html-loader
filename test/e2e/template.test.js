/* eslint-disable */
import path from 'path';
import dom from '../helpers/dom';
import webpack from '../helpers/compiler';

describe('E2E', () => {
  test('Templates', async () => {
    const config = {
      rules: [
        {
          test: /\.html$/,
          use: [
            {
              loader: path.resolve('./src'),
              options: { template: true }
            }
          ]
        },
        {
          test: /\.png$/,
          use: [ 'file-loader' ]
        }
      ]
    };

    const stats = await webpack('e2e/template.js', config);
    const { assets } = stats.compilation;

    const scripts = {
      main: assets['main.js'].source(),
      runtime: assets['runtime.js'].source()
    };

    const { window } = dom([ scripts.runtime, scripts.main ]);

    expect(window.document.body.innerHTML).toMatchSnapshot();
  })
})
