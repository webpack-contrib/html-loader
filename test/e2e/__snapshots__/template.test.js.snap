// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E Templates 1`] = `
"<div id=\\"app\\">
  <!-- HTML Template -->
  Hello World!
  <!-- HTML Import -->
  <div class=\\"import\\"></div>

  <!-- HTML URL -->
  <img src=\\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\">
</div>
"
`;
